metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/bolt/JoinBolt.java

```diff
@@ -89,29 +91,25 @@ public void prepare(Map map, TopologyContext topologyContext, OutputCollector ou
     if (this.maxTimeRetain == null) {
       throw new IllegalStateException("maxTimeRetain must be specified");
     }
-    loader = new CacheLoader<String, Map<String, Tuple>>() {
-      @Override
-      public Map<String, Tuple> load(String key) throws Exception {
-        return new HashMap<>();
-      }
-    };
-    cache = CacheBuilder.newBuilder().maximumSize(maxCacheSize)
-            .expireAfterWrite(maxTimeRetain, TimeUnit.MINUTES).removalListener(new JoinRemoveListener())
-            .build(loader);
+    loader = s -> new HashMap<>();
+    cache = Caffeine.newBuilder().maximumSize(maxCacheSize)
+                         .expireAfterWrite(maxTimeRetain, TimeUnit.MINUTES)
+                         .removalListener(new JoinRemoveListener())
```

Comment:
It seems like we only want notified of a full cache when ERROR logging is set. Is that the case? In the `JoinRemoveListener` we end up doing some work that we probably don't need to do unless ERROR logging is set.  One easy fix would be to only add the "remove listener" if `LOG.isDebugEnabled()`.

Reply:
So, I believe this was intentionally done before this PR (I migrated this to the new caching strategy) and the idea is that if a removal is happening from the join cache under specific circumstances, we want to know about it because a message could be being dropped because the cache is being overwhelmed.  @merrimanr Can you chime in here on the rationale?

Reply:
Yes, it is pre-existing.  We can address at a later time.

I remember now, maxing out this cache causes the Split/Join to fail, which is a major problem for the Split/Join topology.  And this cache here is only for the Split/Join, not the Unified topology.

We should probably look at adding similar logging (only when ERROR enabled) for the other places where we use the cache.  Or just some mechanism to periodically log cache stats.  Anywho, down the road.